src/libsystemd-network/network-internal.c

```diff
@@ -101,11 +102,46 @@ static bool net_condition_test_strv(char * const *patterns, const char *string)
         return has_positive_rule ? match : true;
 }
 
+static int net_condition_test_property(char * const *match_property, sd_device *device) {
+        char * const *p;
+
+        if (strv_isempty(match_property))
+                return true;
+
+        STRV_FOREACH(p, match_property) {
+                _cleanup_free_ char *key = NULL;
+                const char *val, *dev_val;
+                bool invert, v;
+
+                invert = **p == '!';
+
+                val = strchr(*p + invert, '=');
+                if (!val)
+                        return -EINVAL;
+
+                key = strndup(*p + invert, val - *p - invert);
+                if (!key)
+                        return -ENOMEM;
+
+                val++;
+
+                v = device &&
```

Comment:
Hmm, if `device == NULL`, this should return false, no? I.e. maybe change to
```C
   if (strv_isempty(match_property))
         return true;
   if (!device)
         return false;
```
and remove the check on `device` here?

Reply:
If .network or .link file has the following,
```
Property=!KEY=value
```
then I think it should applied to interface which does not have `sd_device` object.
I think it is consistent to
```
Type=!dummy
```
In this case, even if `device` is null, the file is applied to the interface.

Reply:
Oh, right, I didn't take this into account. So if we have a positive match (non-inverted), and device is NULL, the result should be false. If we have a negative match (inverted), and the device is NULL, the result should be true.
So I think something like this needs to be used:
```C
v = device &&
                        sd_device_get_property_value(device, key, &dev_val) >= 0 &&
                        fnmatch(val, dev_val, 0) == 0;
if (invert ? v : !v || !device)
             return false;
```

Reply:
??? `|| !device` is completely redundant, isn't it?

Reply:
Oh, right. I was an idiot. So indeed, this looks ok as is.